plugins/Marketplace/config/config.php

```diff
@@ -1,16 +1,16 @@
 <?php
 
+use Piwik\Http;
 use Psr\Container\ContainerInterface;
 use Piwik\Plugins\Marketplace\Api\Service;
 use Piwik\Plugins\Marketplace\LicenseKey;
 
 return array(
     'MarketplaceEndpoint' => function (ContainerInterface $c) {
-        $domain = 'http://plugins.matomo.org';
-        $updater = $c->get('Piwik\Plugins\CoreUpdater\Updater');
+        $domain = 'https://plugins.matomo.org';
 
-        if ($updater->isUpdatingOverHttps()) {
-            $domain = str_replace('http://', 'https://', $domain);
+        if (!Http::isUpdatingOverHttps()) {
```

Comment:
I think maybe this should always use https unless `force_marketplace_ssl_request` is set to 0, rather than if https isn't supported. This way they would have to explicitly set that config to 0 if they want to use http, rather than falling back to it by default
```suggestion
        if (GeneralConfig::getConfigValue('force_marketplace_ssl_request') === 0) {
```

Reply:
(would also have to add `use Piwik\Config\GeneralConfig` line or similar)